printrun/gl/libtatlin/actors.py

```diff
@@ -315,6 +316,45 @@ def get_next_move(gcode, layer_idx, gline_idx):
         gline_idx = 0
     return None
 
+def interpolate_arcs(glines):
+    prev_gline = None
+    for gline_idx, gline in enumerate(glines):
+        if gline.command == "G2" or gline.command == "G3":
+            rx = gline.i if gline.i is not None else 0
+            ry = gline.j if gline.j is not None else 0
+            r = math.sqrt(rx*rx + ry*ry)
+
+            cx = prev_gline.current_x + rx
+            cy = prev_gline.current_y + ry
+
+            a_start = math.atan2(-ry, -rx)
+            dx = gline.current_x - cx
+            dy = gline.current_y - cy
+            a_end = math.atan2(dy, dx)
+            a_delta = a_end - a_start
+
+            if gline.command == "G3" and a_delta <= 0:
+                a_delta += math.pi * 2
+            elif gline.command == "G2" and a_delta >= 0:
+                a_delta -= math.pi * 2
+
+            # max segment size: 0.5mm, max num of segments: 100
+            segments = math.ceil(abs(a_delta) * r * 2 / 0.5)
+            if segments > 100:
+                segments = 100
+
+            for t in range(segments):
+                a = t / segments * a_delta + a_start
+
+                mid = copy.copy(gline)
+                mid.current_x = cx + math.cos(a) * r
+                mid.current_y = cy + math.sin(a) * r
+                yield (gline_idx, mid)
```

Comment:
Copying and translating G2 Arc commands  and pretending they are lines is neither semantically, nor performance-wise desirable. Would make sense only if we have to fool some foreign interface we do not control. Fortunately the site of consumption of interpolate_arcs is in the same file. 
Here it's seen that the only value/information that we add is tuple(current_x, current_y), which a better candidate for return value
I think the clean way is to add a separate inner iteration loop which eventually expands glines to points, so we have both access to the original gline and the interpolated points, without copying.
The indisputable advantage of the current setup is that it exists and works :)